src/threat_model/rust_client_threat_model.md

```diff
@@ -0,0 +1,134 @@
+
+### Configuration data- AS3
+
+Configuration data passed by the client application to the library. There is some overlap between
+the configuration data and the application identity, as the identity is stored within the client
+library and could thus be considered part of its configuration data.
+
+**Confidentiality** : if known (the authentication token), an attacker could impersonate a specific
+application and execute operations in their name.
+
+**Integrity** : the data should only be modified by the client application as doing otherwise might
+prevent the client application from accessing Parsec functionality.
+
+### Client’s cryptographic keys - AS4
+
+Data stored in the configuration file, read at each reloading of Parsec to instantiate components.
+
+**Confidentiality** : (private) key material is sensitive data and should not be exposed.
+
+**Integrity** : neither public nor private key data should be modifiable by attackers.
+
+**Availability** : the client must always be able to use their cryptographic material.
+
+## Attackers
+
+Each dataflow is analysed from an attacker’s perspective using STRIDE method. Nothing is supposed
+on the different components.
+
+In the following tables are present the type of each possible threat, its description, its
+mitigation status and the assets impacted. A threat can be unmitigated (U), mitigated (M) or
+operational (O). The assumptions context applies for all threats but when one of them is
+particularly relevant, it will be noted with ASUM.
+
+### Attacker “Request to Service” - A1
+
+This attacker uses the inter-process endpoint, presumably created by the service, to interfere with
+the communication between client and service.
+
+|   | Description                                                                                                                                       | Mitigation | Assets        |
+|---|---------------------------------------------------------------------------------------------------------------------------------------------------|------------|---------------|
+| S | An attacker impersonates the service by establishing an IPC endpoint that spoofs the one used by Parsec to exfiltrate any data found in requests. | O-0, U-1   | AS1, AS2, AS4 |
+| T | The attacker modifies the data sent or received by the client while it is in transit through the IPC.                                             | M-0, U-1   | AS2, AS4      |
+| R | The service denies having sent responses to or received some requests from the client.                                                            | U-3        |               |
+| I | An attacker obtains requests sent by the client by sniffing the IPC traffic or spoofing the endpoint.                                             | M-0, U-1   | AS2, AS4      |
+| D | An attacker could remove the IPC endpoint or block traffic through it.                                                                            | O-1, U-1   | AS4           |
```

Comment:
As a DoS there is also the possibility that the attacker spams the service with request preventing our trusted client to have their one go through.